pkg/util/log_support.go

```diff
@@ -0,0 +1,347 @@
+type TextFormatter struct {
+	// Set to true to bypass checking for a TTY before outputting colors.
+	ForceColors bool
+
+	// Force disabling colors.
+	DisableColors bool
+
+	// Disable timestamp logging. useful when output is redirected to logging
+	// system that already adds timestamps.
+	DisableTimestamp bool
+
+	// Enable logging the full timestamp when a TTY is attached instead of just
+	// the time passed since beginning of execution.
+	FullTimestamp bool
+
+	// TimestampFormat to use for display when a full timestamp is printed
+	TimestampFormat string
+
+	// The fields are sorted by default for a consistent output. For applications
+	// that log extremely frequently and don't use the JSON formatter this may not
+	// be desired.
+	DisableSorting bool
+
+	// QuoteEmptyFields will wrap empty fields in quotes if true
+	QuoteEmptyFields bool
+
+	// Whether the logger's out is to a terminal
+	isTerminal bool
+
+	sync.Once
+}
+
+// LogSetUp sets up UserOut and log loggers as needed by ddev
+func LogSetUp() {
+	// Use stdout instead of stderr for all user output
+	log.SetOutput(os.Stdout)
+	UserOut.Out = os.Stdout
+
+	if !JSONOutput {
+		UserOut.Formatter = UserOutFormatter
+	} else {
+		UserOut.Formatter = &JSONFormatter{}
+	}
+
+	UserOutFormatter.DisableTimestamp = true
+	// Always use log.DebugLevel for UserOut
+	UserOut.Level = log.DebugLevel // UserOut will by default always output
+
+	// But we use custom DRUD_DEBUG-settable loglevel for log
+	logLevel := log.InfoLevel
+	drudDebug := os.Getenv("DRUD_DEBUG")
+	if drudDebug != "" {
+		logLevel = log.DebugLevel
+	}
+	log.SetLevel(logLevel)
+}
+
+func (f *TextFormatter) init(entry *log.Entry) {
+	if entry.Logger != nil {
+		f.isTerminal = f.checkIfTerminal(entry.Logger.Out)
+	}
+}
+
+func (f *TextFormatter) checkIfTerminal(w io.Writer) bool {
+	switch v := w.(type) {
+	case *os.File:
+		return terminal.IsTerminal(int(v.Fd()))
+	default:
+		return false
+	}
+}
+
+// Format renders a single log entry. A key named "raw" is discarded here as it's for json consumption.
+func (f *TextFormatter) Format(entry *log.Entry) ([]byte, error) {
+	var b *bytes.Buffer
+	keys := make([]string, 0, len(entry.Data))
+	for k := range entry.Data {
+		// Discard key named "raw" as it's for json consumption only.
+		if k != "raw" {
+			keys = append(keys, k)
+		}
+	}
+
+	if !f.DisableSorting {
+		sort.Strings(keys)
+	}
+	if entry.Buffer != nil {
+		b = entry.Buffer
+	} else {
+		b = &bytes.Buffer{}
+	}
+
+	prefixFieldClashes(entry.Data)
+
+	f.Do(func() { f.init(entry) })
+
+	isColored := (f.ForceColors || f.isTerminal) && !f.DisableColors
+
+	timestampFormat := f.TimestampFormat
+	if timestampFormat == "" {
+		timestampFormat = defaultTimestampFormat
+	}
+	if isColored {
+		f.printColored(b, entry, keys, timestampFormat)
+	} else {
+		if !f.DisableTimestamp {
+			f.appendKeyValue(b, "time", entry.Time.Format(timestampFormat))
+		}
+		//f.appendKeyValue(b, "level", entry.Level.String())
```

Comment:
nit, cleanup - this and a couple lines later

Reply:
Fixed in fa6049a0